system/default_ad_api_helpers/src/initial_pose_adaptor.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "initial_pose_adaptor.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace default_ad_api_helpers
+{
+template <class ServiceT>
+using Future = typename rclcpp::Client<ServiceT>::SharedFuture;
+
+std::array<double, 36> GetCovarianceParameter(rclcpp::Node * node, const std::string & name)
+{
+  const auto vector = node->template declare_parameter<std::vector<double>>(name);
```

Comment:
[small question] Do you need this `template` word here?

Reply:
It was unnecessary, removed  https://github.com/autowarefoundation/autoware.universe/pull/1431/commits/b94a6d807c31cfd619b020d31fd6d464d18c98d9. In the my first implementation, this function takes template argument `NodeT * node`, and template keyword is specified because the compiler doesn't know if the declare_parameter function is a normal function or a template function.

Reply:
I see, thanks!